include/glow/ExecutionEngine/ExecutionEngine.h

```diff
@@ -78,34 +78,34 @@ class ExecutionEngine final {
   void save(CompilationMode mode, Function *F, llvm::StringRef outputDir,
             llvm::StringRef networkName);
 
-  /// Runs a single execution of the function. This method updates the variables
-  /// in \p nodes with the tensor content values \p inputs.
-  void run(llvm::ArrayRef<Variable *> vars, llvm::ArrayRef<Tensor *> inputs);
-
-  /// Runs \p iterations iterations of the function. The method updates a local
-  /// counter and future invocations of this method continue running iterations
-  /// of the batch at the next available slice.
-  /// The method updates the variables in \p vars with the tensors \p inputs.
-  void runBatch(size_t iterations, llvm::ArrayRef<Variable *> vars,
-                llvm::ArrayRef<Tensor *> inputs);
-
-private:
-  /// Update the inputs for all variables \p vars with data from the inputs \p
-  /// inputs at offset \p sampleIdx. Then perform a run of the network.
-  void updateInputsAndRunNetwork(llvm::ArrayRef<Variable *> vars,
-                                 llvm::ArrayRef<Tensor *> inputs,
-                                 size_t sampleIdx);
-
-  /// Update the content of the tensor \p v with some slices that from \p input.
-  /// The data starts at slice \p sampleIdx and wraps around until the
-  /// data in \p v is filled. All dimensions, except for the first (batch)
-  /// dimension must be identical.
-  void loadValueFromTensorSlice(Variable *v, Tensor *input, size_t sampleIdx);
-
-  // Update the content of the tensor \p v with \p input.
-  void loadValueFromTensor(Variable *v, Tensor *input);
+  /// Runs a single execution of the function.
+  void run();
 };
 
+//===----------------------------------------------------------------------===//
+//         Helper methods for running the execution engine.
+//===----------------------------------------------------------------------===//
+
+/// This method updates the variables in \p nodes with the tensor content
+/// values \p inputs.
+void updateVariables(llvm::ArrayRef<Variable *> vars,
+                     llvm::ArrayRef<Tensor *> inputs);
+
+/// Update the content of the tensors \p vars with some slices that from \p
+/// inputs. The data starts at slice \p sampleIdx and wraps around until the
+/// data in \p v is filled. All dimensions, except for the first (batch)
+/// dimension must be identical.
+void updateVariablesFromBatch(llvm::ArrayRef<Variable *> vars,
+                              llvm::ArrayRef<Tensor *> inputs,
+                              size_t sampleIdx);
+
+/// Runs \p iterations iterations of the function. The method updates a local
+/// counter and future invocations of this method continue running iterations
+/// of the batch at the next available slice.
+/// The method updates the variables in \p vars with the tensors \p inputs.
+void runBatch(ExecutionEngine &EE, size_t iterations,
```

Comment:
Is the part with the local counter still true?
Also, while you are here, the comment would benefit from more love :).
After reading it over and over, I am not sure how the iterations are applied to the batch size.

Reply:
I updated the comment and got rid of the global variable. 

/// Runs \p iterations iterations of the compiled function. The method updates a
/// global counter and future invocations of this method continue running
/// iterations of the batch at the next available slice.
///
/// The method updates the variables in \p vars with the tensors \p inputs. The
/// shape of the slice has to be identical to the shape of slices in the batch.
/// All dimensions, except for the first (batch) dimension must be identical.
///
/// The variable \p sampleCounter is consumed and updated by the function. This
/// variable records the number of samples that were consumed by the network in
/// previous iterations. The next input to be loaded is
/// (sampleCounter % batchsize).